syntax = "proto3";
package DomainServices.OfferService;
option csharp_namespace = "DomainServices.OfferService.Contracts";

import "Identity.proto";
import "ModificationStamp.proto";
import "NullableGrpcDate.proto";
import "Parameters/MortgageOffer.proto";

message SimulateMortgageRequest {
	string ResourceProcessId = 1;
	MortgageOfferBasicParameters BasicParameters = 2;
	MortgageOfferSimulationInputs SimulationInputs = 3;
	bool IsCreditWorthinessSimpleRequested = 4;
	MortgageOfferCreditWorthinessSimpleInputs CreditWorthinessSimpleInputs = 5;
	repeated cis.types.Identity Identities = 6;
	cis.types.NullableGrpcDate ValidTo = 7;
}

message SimulateMortgageResponse {
	int32 OfferId = 1;
	string ResourceProcessId = 2;
	cis.types.ModificationStamp Created = 3;
	MortgageOfferBasicParameters BasicParameters = 4;
	MortgageOfferSimulationInputs SimulationInputs = 5;
	MortgageOfferSimulationResults SimulationResults = 6;
	MortgageOfferAdditionalSimulationResults AdditionalSimulationResults = 7;
	MortgageOfferCreditWorthinessSimpleResults CreditWorthinessSimpleResults = 8;
}
